src/cr/cube/dimension.py

```diff
@@ -586,6 +589,14 @@ def _element_dicts(self):
             else self._type_dict["elements"]
         )
 
+    def _element_id(self, element_dict):
+        """Returns the element identifier given its dict."""
+        if self._dimension_type in DT.ARRAY_TYPES:
```

Comment:
It would be much nicer to have explicit decision making here (like based on that mentioned `"key"` argument in the transforms).